File: 5Days-SoftwareArchitecture/typesript/Day3/Complexity/high_comp.js
var Logic = /** @class */ (function () {
    function Logic() {
    }
    Logic.prototype.checkFlat = function (x) {
        if (x <= 0)
            return;
        if (x >= 100)
            return;
        if (x % 2 !== 0)
            return;
        console.log("Even Positive Small");
    };
    return Logic;
}());
var logic = new Logic();
logic.checkFlat(10);

File: 5Days-SoftwareArchitecture/Java/Day3/Complexity/high_comp.java
class Logic {
    // POOR readability due to deep nesting
    // Cyclomatic Complexity = 4
    public void checkNested(int x) {
        if (x > 0) {
            if (x < 100) {
                if (x % 2 == 0) {
                    System.out.println("Even Positive Small");
                }
            }
        }
    }
}
class high_comp {
    public static void main(String[] args)
    {
        var logic = new Logic();
        logic.checkNested(10);
      
    }
}
